ote/src/clj/ote/tasks/gtfs.clj

```diff
@@ -105,6 +106,34 @@
                  (log/warn e "Change detection failed for service " service-id)))
              (log/info "Detection completed for service: " service-id))))))))
 
+(defn recalculate-detected-changes-count
+"Change detection (detect-new-changes-task) stores detected changes. These changes will expire after some period of time.
+All changes occur on a certain day and they will get old. Transit-changes page shows how many changes there are and
+if we don't recalculate them every night expired changes will be show also. So in this fn we recalculate change counts again using
+different-week-date value and skip all expired changes."
+  [db]
+  (let [upcoming-changes (upcoming-changes db)
+        _ (log/info "Detected change count recalculation started!")]
+    ;; Loop upcoming-changes
+    (dotimes [i (count upcoming-changes)]
```

Comment:
Ok like this of courser,
food for thought: would a more functional approach be to loop upcoming-changes and per iteration take `first` and recur `rest`?
loop syntax is always a bit verbose :/

Reply:
I'm not sure if this is such a bad option because to enable loop with index number there will be extra code.

Reply:
loop wouldn't use index, it would use only `first` and `rest`.
But anyways...